identity/clients/registry_test.go

```diff
@@ -0,0 +1,38 @@
+package clients
+
+import "testing"
+
+func TestRedirectUriWithDynamicPort(t *testing.T) {
+	redirectURIs := []struct {
+		uri       string
+		shallFail bool
+	}{
+		{"http://localhost:12345", false},
+		{"http://localhost:12345/callback", false},
+		{"http://localhost", false},
+		{"custom://callback.example.net", false},
+		{"http://localhost:12345/callback-disallowed", true},
+		{"http://localhost.example.net/callback", true},
+	}
+
+	registry, _ := NewRegistry(nil, nil, "", nil)
+	clientRegistration := ClientRegistration{
+		ID:              "native",
+		Secret:          "secret",
+		Trusted:         true,
+		TrustedScopes:   nil,
+		Insecure:        false,
+		Dynamic:         false,
+		ApplicationType: "native",
+		RedirectURIs:    []string{"http://localhost", "http://localhost/callback", "custom://callback.example.net"},
```

Comment:
Nice to see unit tests! Would you please also add a registered URL with a port, to ensure that this also works.